aten/src/ATen/native/TensorShape.cpp

```diff
@@ -153,6 +153,30 @@ Tensor &as_strided_(Tensor& self, IntList size, IntList stride) {
   return at::as_strided_(self, size, stride, self.storage_offset());
 }
 
+Tensor narrow_copy(const Tensor& self, int64_t dim, int64_t start, int64_t length){
+  if (self.type().is_sparse()) {
+    AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
+    LongTensor indices = self._indices();
+    int64_t nnz = indices.size(1);
+    int64_t dims = indices.size(0);
+    int64_t end = start+length;
+    
+    std::vector<int64_t> newSizes = self.sizes().vec();
+    newSizes[dim]=length;
+    
+    Tensor narrowDim = at::zeros_like(indices[dim]);
+    narrowDim.copy_(indices[dim]);
+    Tensor mask = (narrowDim >= start).__and__((narrowDim < end));
+    
+    indices[dim] = indices[dim].add(-start);
+    Tensor newValues = self._values().masked_select(mask);
+    Tensor newIndices = indices.masked_select(mask).view({dims, -1});
+    return self.type().sparse_coo_tensor(newIndices, newValues, newSizes);
+  }else{
+    return self.clone().narrow(dim,start,length);
```

Comment:
You want it the other way: clone after the narrow. That will reduce storage space.